app/tasksmodule/datachecks.py

```diff
@@ -0,0 +1,178 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tasks to check data.
+
+These tasks don't modify data, they just double-check things that are already
+expected to be true.
+"""
+
+import datetime
+import time
+
+import django.http
+from google.appengine import runtime
+from google.appengine.api import datastore_errors
+from google.appengine.api import taskqueue
+
+import tasksmodule.base
+import model
+import utils
+import views.base
+
+
+class DatacheckException(Exception):
+    """An exception raised when a datacheck tasks finds a problem."""
+    pass
+
+
+class DatachecksBaseTask(tasksmodule.base.PerRepoTaskBaseView):
+
+    def get_params(self):
+        return views.base.read_params(
+            super(DatachecksBaseTask, self).get_params(),
+            self.request,
+            post_params={'cursor': utils.strip})
+
+    def schedule_task(self, repo, **kwargs):
+        name = '%s-%s-%s' % (repo, self.BASE_NAME, int(time.time()*1000))
+        path = self.build_absolute_path('/%s/tasks/%s' % (repo, self.TASK_PATH))
+        cursor = kwargs.get('cursor', '')
+        # TODO(nworden): figure out why setting task_retry_limit isn't working
+        retry_options = taskqueue.taskqueue.TaskRetryOptions(task_retry_limit=1)
+        taskqueue.add(name=name, method='POST', url=path,
+                      queue_name='datachecks', retry_options=retry_options,
+                      params={'cursor': cursor})
+
+    def alert(self, msg):
+        """Alerts developers of an error.
+
+        For now, we just raise an exception. It will show up in Stackdriver.
+        """
+        raise DatacheckException(msg)
+
+
+class PersonDataValidityCheckTask(DatachecksBaseTask):
+    """Checks that person records have valid data."""
+
+    BASE_NAME = 'person_data_validity_check'
+    TASK_PATH = 'check_person_data_validity'
+
+    def _check_person(self, person):
+        if not person.entry_date:
+            self.alert('A person record is missing an entry_date value.')
+        if not person.original_creation_date:
+            self.alert(
+                'A person record is missing an original_creation_date value.')
+        if (person.author_email and
+                not utils.validate_email(person.author_email)):
+            self.alert('A person record has an invalid author_email value.')
+
+    def post(self, request, *args, **kwargs):
+        del request, args, kwargs  # unused
+        q = model.Person.all(filter_expired=False).filter(
+            'repo =', self.env.repo)
+        cursor = self.params.get('cursor', '')
+        if cursor:
+            q.with_cursor(cursor)
+        try:
+            for person in q:
+                next_cursor = q.cursor()
+                self._check_person(person)
+                cursor = next_cursor
```

Comment:
seems like dark magic to me, but I'll take your word for it

Reply:
Heh, we do this because jobs can run over GAE's task time limit. The cursor variable should always point to a record at or before the last record we checked, so that the next task scheduled picks up at or before where the last one left off and we ensure that every record gets checked.